### PR TITLE
Feat/fly 2545

### PR DESCRIPTION
**Pull request by Hakob23**
## What

Adds the two deadline-driven refund exits on PegOutEscrow and wires them to slash hooks:

- **`refundOnNoClaim`**: after the claim window (`depositDateLimit`), anyone can refund the user from escrow, set `REFUNDED`, and attempt `globalSlash` (user payout does not depend on slash success; failures emit `GlobalSlashSkipped`).
- **`onSettlement`**: PegOutContract notifies escrow when a claimed peg-out finishes as `REFUNDED` (also accepts `FULFILLED` for later settlement wiring).
- **`refundUserPegOut`**: when PegOutEscrow is set, loads the `CLAIMED` quote from escrow, refunds the user, individual-slashes via existing `slashPegOutCollateral`, then calls `onSettlement(REFUNDED)`. Legacy `depositPegOut` path remains when escrow is unset (`QuoteNotFound` on replay).
- **`ICollateralManagement.globalSlash`**: role-gated stub that reverts `GlobalSlashNotImplemented` until the real proportional slash lands.

Acceptance-focused PegOutEscrow tests cover: blocked before deadline, permissionless refund after deadline + global slash attempt, cancel does not call global slash, claimed-unfulfilled refund + individual slash + late replay revert.

## Why

With commit-first peg-out, an unclaimed request past the claim deadline and a claimed-but-unfulfilled request past fulfillment expiry must refund the user without leaving funds stuck, while keeping cancellation (no slash) distinct from deadline refunds (network / LP fault → slash hooks).

## Type of Change

- [ ] Bug fix (non-breaking change)
- [x] New feature (non-breaking change)
- [ ] Breaking change
- [ ] Refactor (no intended behavior change)
- [ ] Security hardening
- [ ] Tests only
- [ ] Build/CI/Tooling
- [ ] Documentation

## Affected Areas

- [ ] PegIn flow (`registerPegIn`, `callForUser`, quote validation)
- [x] PegOut flow (`depositPegout`, `refundPegOut`, `refundUserPegOut`)
- [ ] Liquidity Provider lifecycle (register/update/status/resign/collateral/withdraw)
- [ ] Pause/operational controls
- [ ] Quote hashing/signature logic
- [ ] Bitcoin tx / PMT / merkle validation
- [ ] Deployment or upgrade scripts / Makefile targets
- [ ] CI workflows (`ci`, `fuzz`, `formal`, `slither`, `codeql`)
- [ ] Docs

Also touches: `PegOutEscrow`, `ICollateralManagement` / `CollateralManagement` (globalSlash stub), collateral mock, PegOutEscrow unit tests.

## Risk & Security Review

- [x] Access control and authorization paths reviewed
- [x] Reentrancy and external call paths reviewed
- [x] Storage layout / upgrade safety reviewed (if upgradeable code changed)
- [x] Time/block/confirmation assumptions reviewed
- [ ] BTC tx output/index assumptions reviewed (if pegout validation touched)
- [x] No secrets or sensitive values added

Notes:
- `refundOnNoClaim` / `refundUserPegOut` are permissionless after their deadlines; payout goes to the snapshotted refund address.
- Global slash is try/catch’d so a stub/missing-role/revert cannot block the user refund.
- Escrow soft-skips in `_notifyEscrowSettlement` when unset or not `CLAIMED` (legacy coexistence); TODOs mark hardening after `depositPegOut` retirement.
- `globalSlash` body is intentionally unimplemented (stub only).

## Test Plan

List what you ran locally and the outcome:

- [x] `npm run lint:sol` (error fixed; remaining warnings pre-existing / non-blocking)
- [x] `npm run lint:ts`
- [ ] `npm run compile`
- [x] `npm test`
- [ ] `npm run test:fuzz` (if logic/state transitions changed)
- [ ] `npm run test:invariant` (if invariant-sensitive logic changed)
- [ ] `npm run test:formal` (if formal-verification-sensitive logic changed)
- [ ] `npm run test:integration` (if integration paths changed)
- [ ] `npm run test:e2e` (if deployment/ownership flows changed)

Additional notes on test coverage, edge cases, and any tests intentionally skipped:

```bash
forge test --match-path 'test/pegout-escrow/*'
forge test --match-path 'test/pegout/UserRefund.t.sol'
```

Both green. Coverage is acceptance-oriented for the two deadline exits + cancel vs global-slash distinction; full proportional `globalSlash` math and SPV `FULFILLED` notify are out of scope.

## Deployment & Ops Impact

- [ ] No deployment impact
- [x] Requires deployment
- [x] Requires upgrade
- [ ] Env/config changes required (`.env`, network RPCs, verification, etc.)
- [ ] Runbook/update instructions added to PR description

Upgradeable contracts touched: PegOutEscrow, PegOutContract, CollateralManagement. After deploy, PegOutEscrow must be wired (`setPegOutEscrow` / escrow deps) and should hold `COLLATERAL_SLASHER` before a real `globalSlash` body is enabled (today the stub reverts and is skipped).

## Related Issues

- Jira: [FLY-2545](https://rsklabs.atlassian.net/browse/FLY-2545)

## Reviewer Notes

- Focus on fund custody: no-claim refund pays from **escrow**; claimed-unfulfilled refund pays from **PegOutContract**.
- Three exits: cancel (no slash), no-claim refund (global slash attempt), claimed timeout (`slashPegOutCollateral` on responsible LP).
- `refundUserPegOut` branches on `address(_pegOutEscrow) == 0` (legacy) vs set (commit-first); `QuoteAlreadyCompleted` is only on the commit-first branch so legacy replay stays `QuoteNotFound`.
- Do not expect a working global slash distribution yet — interface + stub + try/catch wiring only.
- `onSettlement` accepts `FULFILLED` | `REFUNDED`, but this PR only notifies `REFUNDED` from `refundUserPegOut` (SPV path notify deferred).

### CODE REVIEW

File: src/PegOutContract.sol [hunk 1]
```diff
@@ -309,9 +309,20 @@ contract PegOutContract is
 
     /// @inheritdoc IPegOut
     function refundUserPegOut(bytes32 quoteHash) external nonReentrant whenNotHardPaused override {
-        Quotes.PegOutQuote memory quote = _pegOutQuotes[quoteHash];
-
-        if (quote.lbcAddress == address(0)) revert Flyover.QuoteNotFound(quoteHash);
+        // TODO: after retiring depositPegOut, require escrow (revert PegOutEscrowNotSet) and drop the legacy branch.
+        Quotes.PegOutQuote memory quote;
+        if (address(_pegOutEscrow) == address(0)) {
```
Comment (Luisfc68) [inline src/PegOutContract.sol:314]: no need to maintain the legacy path, we do not plan to support both versions of the protocolo at the same time.

File: src/PegOutContract.sol [hunk 2]
```diff
@@ -309,9 +309,20 @@ contract PegOutContract is
...
+            // Legacy depositPegOut: replay after refund surfaces QuoteNotFound (quote body deleted).
+            quote = _pegOutQuotes[quoteHash];
+            if (quote.lbcAddress == address(0)) revert Flyover.QuoteNotFound(quoteHash);
+        } else {
+            // Commit-first: quote body lives on PegOutEscrow while CLAIMED.
+            if (_isQuoteCompleted(quoteHash)) revert QuoteAlreadyCompleted(quoteHash);
+            if (_pegOutEscrow.getPegOutState(quoteHash) != IPegOutEscrow.EscrowedPegOutState.CLAIMED) {
```
Comment (Luisfc68) [inline src/PegOutContract.sol:321]: IMO there's no need of this, you can maintain the current flow, if the contract doesn't know about the quote revert, and if it knows continue with the current process, if the contract is aware then it must have been claimed in the escrow, here I think we're mixing what should be living on the contract with what should be living in the escrow, finally, you should not read the quote from there, by this point you're supposed to have it in the contract, so read it from here

File: src/PegOutContract.sol
Comment (Luisfc68) [inline src/PegOutContract.sol:1]: To summarize my opinion on this file: the only change in `refundUserPegOut` should be the addition of `_notifyEscrowSettlement`

File: src/PegOutEscrow.sol [hunk 3]
```diff
@@ -212,15 +212,42 @@ contract PegOutEscrow is
...
+        if (block.timestamp <= q.depositDateLimit) {
+            revert ClaimWindowOpen(q.depositDateLimit);
+        }
+
+        uint256 payout = q.value + q.callFee + q.gasFee;
+        _terminate($, requestHash, EscrowedPegOutState.REFUNDED);
+        emit PegOutRefundedOnNoClaim(requestHash, q.rskRefundAddress, payout);
+
+        // If globalSlash reverts (stub / missing role / no eligible LPs), user still refunded.
+        if (address($.collateralManagement) == address(0)) revert CollateralManagementNotSet();
```
Comment (Luisfc68) [inline src/PegOutEscrow.sol:229]: shouldn't this check be at the beginning?

**Review by Luisfc68 — CHANGES_REQUESTED** [review]

**Comment by Hakob23** [comment]
Fixes are applied in https://github.com/rsksmart/liquidity-bridge-contract/pull/549
